### PR TITLE
Vb 3017 - Improve alerts visibility

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this PR do?
It Filters alerts to only handle and return supported codes. These are defined in a list which can be found on the JIRA parent ticket.

## JIRA Link
https://dsdmoj.atlassian.net/browse/VB-3017
https://dsdmoj.atlassian.net/browse/VB-4921

### CODE REVIEW

**Review by dhirajshettymoj — COMMENTED** [review]
minor comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerAlertsUpdatedNotificationInfo.kt [hunk 1]
```diff
@@ -9,8 +9,8 @@ data class PrisonerAlertsUpdatedNotificationInfo(
   val nomsNumber: String,
 
   @JsonProperty("alertsAdded")
-  val alertsAdded: List<String>,
+  var alertsAdded: List<String>,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerAlertsUpdatedNotificationInfo.kt:12]: should this and below variable be renamed to supportedAlerts or something similar?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/AlertsService.kt [hunk 2]
```diff
@@ -4,24 +4,30 @@ import org.slf4j.Logger
...
   /**
    * Gets all alerts and filters them to only return active. Maps the alert object to only capture the alert codes as a list<String>.
    */
-  fun getPrisonerActiveAlertCodes(prisonerId: String): List<String> {
-    val prisonerAlerts = alertsApiClient.getPrisonerAlerts(prisonerId)
-    logger.info("Retrieved $prisonerAlerts alerts for $prisonerId")
-    prisonerAlerts?.let {
-      return prisonerAlerts.content.filter { it.active }.map { it.alertCode.code }.toList()
-    } ?: throw NotFoundException("getPrisonerActiveAlertCodes - alertsApiClient failed to get alerts for prisoner $prisonerId")
+  fun getPrisonerAlertCodes(prisonerId: String): List<AlertResponseDto> {
```
[the remark this answers is not in the document]
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/AlertsService.kt]: Added an isActive flag to the alerts-api call directly, so no isActive logic is needed anymore.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/PrisonerProfileClient.kt [hunk 3]
```diff
@@ -47,7 +49,7 @@ class PrisonerProfileClient(
         val visitBalances = if (prisonerProfileMonos.t3.isEmpty) null else prisonerProfileMonos.t3.get()
         val prisonerBookingSummary = prisonerProfileMonos.t4.content.firstOrNull()
         val visits = prisonerProfileMonos.t5.content.map { visitDto -> VisitSummaryDto(visitDto = visitDto) }
-        val prisonerAlerts = prisonerProfileMonos.t6.content.map { alertResponse -> AlertDto(alertResponse) }
+        val prisonerAlerts = alertsService.filterSupportedAlerts(prisonerProfileMonos.t6).map { alertResponse -> AlertDto(alertResponse) }
```
[the remark this answers is not in the document]
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/PrisonerProfileClient.kt]: Added a predicate to the companion object as requested (thanks for the assistance with this!)

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**